src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -365,17 +365,21 @@ uct_dc_mlx5_dci_pool_init_dci(uct_dc_mlx5_iface_t *iface, uint8_t pool_index,
         num_channels = iface->tx.num_dci_channels;
     }
 
-    status = uct_dc_mlx5_iface_create_dci(iface, dci_index, num_channels);
+    status = uct_dc_mlx5_iface_create_dci(iface, dci_index, num_channels, &dci);
     if (status != UCS_OK) {
         ucs_error("iface %p: failed to create dci %u at pool %u", iface,
                   dci_index, pool_index);
         goto err;
     }
 
-    dci             = uct_dc_mlx5_iface_dci(iface, dci_index);
     dci->path_index = pool->config.path_index;
     dci->pool_index = pool_index;
 
+    ucs_assertv_always(!uct_dc_mlx5_is_dci_valid(
```

Comment:
moved this assert outside of uct_dc_mlx5_iface_create_dci, since uct_dc_mlx5_iface_create_dci is also called from uct_dc_mlx5_iface_init_dcis_array to create a dumy dci to count tx.bb